src/App.js

```diff
@@ -40,41 +40,69 @@ export default class App extends Component {
   state = {
     isAuthenticated: netlifyIdentity.currentUser() != null,
     user: netlifyIdentity.currentUser(),
-    profile: null
+    profile: null,
+    groups: null,
+    jwt: null,
   }
   constructor(props) {
     super(props);
     this.authenticate = this.authenticate.bind(this);
     this.signout = this.signout.bind(this);
+    this.updateGroups = this.updateGroups.bind(this);
   } 
   authenticate() {
     netlifyIdentity.open();
     netlifyIdentity.on('login', user => {
-      this.setState({
-        isAuthenticated: true,
-        user: user
-      })
-    });
+       user.jwt().then((jwt) => {
+         return api.readGroups(jwt)
+       }).then((groups) => {
+    	 this.setState({
+    	   groups: groups,
+           isAuthenticated: true,
+           user: user
+         })
+       })      
+    })
   }
   signout() {
     netlifyIdentity.logout();
     netlifyIdentity.on('logout', () => {
       this.setState({
         isAuthenticated: false,
-        user: null
+        user: null,
+        groups: null
       })
     });
   }
+  updateGroups() {
+    const { jwt } = this.state
+    api.readGroups(jwt)
+    .then((groups) => {
+    	 this.setState({
+    	   groups: groups
+       })
+    })
+  }
```

Comment:
@amn41 let's put this in a helper function and call it from the component directly instead of passing it in the props.